src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -338,11 +338,15 @@ private function set($type, $id)
      */
     private function createAutowiredDefinition(\ReflectionClass $typeHint, $id)
     {
-        if (isset($this->ambiguousServiceTypes[$typeHint->name])) {
+        if (isset($this->ambiguousServiceTypes[$typeHintName = $typeHint->name])) {
```

Comment:
Why is necessary to create the `$typeHintName` variable? It makes the code harder to read for no benefit over using the prop directly.